src/components/ec-radio-btn/ec-radio-btn.vue

```diff
@@ -0,0 +1,298 @@
+          'ec-radio-btn__icon-wrapper--checked-and-disabled': isDisabled && isChecked,
+        }"
+      >
+        <div
+          class="ec-radio-btn__icon-wrapper-inner"
+        >
+
+          <ec-icon
+            class="ec-radio-btn__icon"
+            :class="{
+              'ec-radio-btn__icon--checked': isChecked && !isDisabled,
+              'ec-radio-btn__icon--checked-and-disabled': isDisabled && isChecked
+            }"
+            name="rounded-notification"
+            :size="20"
+          />
+        </div>
+      </div>
+
+      <div
+        v-if="hasLabel"
+        data-test="ec-radio-btn__text-wrapper"
+        class="ec-radio-btn__text-wrapper"
+        :class="{
+          'ec-radio-btn__text-wrapper--is-single-line': isTextInline,
+        }"
+      >
+        <label
+          :for="name"
+          class="ec-radio-btn__label"
+          :class="{
+            'ec-radio-btn__label--disabled': isDisabled,
+          }"
+          :title="isTextInline ? label : undefined"
+          data-test="ec-radio-btn__label"
+        >
+          <slot name="label">{{ label }}</slot>
+        </label>
+
+        <p
+          v-if="hasDescription"
+          class="ec-radio-btn__description"
+          :class="{
+            'ec-radio-btn__description--is-single-line': isTextInline,
+            'ec-radio-btn__description--disabled': isDisabled,
+          }"
+          :title="isTextInline ? description : undefined"
+          data-test="ec-radio-btn__description"
+        >
+          <slot name="description">{{ description }}</slot>
+        </p>
+      </div>
+    </div>
+
+    <div
+      :id="errorId"
+      v-if="hasErrorMessage"
+      class="ec-radio-btn__error-text"
+      data-test="ec-radio-btn__error-text"
+    >
+      <slot name="error-message">{{ errorMessage }}</slot>
+    </div>
+  </div>
+
+</template>
+
+<script setup lang="ts">
+import {
+  computed, ref, useSlots,
+} from 'vue';
+
+import { getUid } from '../../utils/uid';
+import EcIcon from '../ec-icon';
+import type { RadioButtonEvents } from './types';
+import { RadioButtonEvent } from './types';
+
+interface RadioButtonProps {
+  value: string,
+  modelValue?: string,
+  label?: string,
+  description?: string,
+  isDisabled?: boolean,
+  isTextInline?: boolean,
+  name?: string,
+  errorMessage?: string,
+  hasError?: boolean
+}
+
+const props = defineProps<RadioButtonProps>();
+
+const uid = getUid();
+const slots = useSlots();
+
+const id = `ec-radio-btn-${uid}`;
+const errorId = `error-${id}`;
+const labelId = `label-${id}`;
+const descriptionId = `description-${id}`;
+const inputRadioRef = ref();
+
+const emit = defineEmits<{(e: 'update:modelValue', value: RadioButtonEvents[RadioButtonEvent.UPDATE_MODEL_VALUE]): void,
+}>();
+
+const isChecked = computed(() => props.modelValue === props.value);
+const isFocused = ref(false);
+
+const hasLabel = computed(() => (!!props.label || !!slots.label));
```

Comment:
```suggestion
const hasLabel = computed(() => (props.label || slots.label));
```

Reply:
the esclamation marks are needed to convert the string or the undefined into a boolean. this is what i get without them `const hasLabel: ComputedRef<string | Slot | undefined>`